### ISSUE TITLE
Loading in of large BAM files

### ISSUE THREAD

**Issue by jdcla** [wheretrue/biobear#133]
Hello, 

I have an issue opened that seems to be related to BioBear handling of large `.bam` files. The stackframe points to the use of `polars-u64-idx` as a potential solution.

Last line stack frame `pyo3_runtime.PanicException: polars' maximum length reached. Consider installing 'polars-u64-idx'.: TryFromIntError(())`

Issue opened by a user on my github: https://github.com/TRISTAN-ORF/RiboTIE/issues/8

Link to code line using biobear: https://github.com/TRISTAN-ORF/transcript_transformer/blob/235fde98f0719ed86f244a0853d3b9c8f4e9edce/transcript_transformer/data.py#L132

Furthermore, today I had transcript-transformer hang also on the loading step of the `.bam` file for a medium sized file (4GB). Could be related.

**Comment by tshauck** [comment]
Thanks for sharing this! I'll have a look and follow up.

**Comment by tshauck** [comment]
I was able to repro what I think is the issue causing upstream in Exon. Once fixed, I was able to run a query on a 16G file from 1000 genomes.

The Exon release is matriculating, but once I've had a chance to check biobear, I'll do a release here.

I think I also see an area for improvement performance wise I hope to release, but not sure how impactful it is yet. Generally w/ Polars, if I also made it possible to return a LazyFrame, would that be of use to you?
